### PR TITLE
add some metrics.

### PR DESCRIPTION
**Pull request by huachaohuang**
ref #294 #295

### CODE REVIEW

File: pd-client/metrics.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	for name := range pdpb.CommandType_value {
+		types[name] = convertName(name)
+	}
+
+	return types
+}
+
+// convertName converts variable name to a linux type name.
+// Like `AbcDef -> abc_def`.
+func convertName(str string) string {
```
Comment (siddontang) [inline pd-client/metrics.go]: seem pd-server has same function ?

Comment (huachaohuang) [inline pd-client/metrics.go]: Yes, but pd-client doesn't depend on pd-server.

Comment (qiuyesuifeng) [inline pd-client/metrics.go]: Maybe we can remove pd-client outside as a new repo. 
How do you think that?

File: pd-client/rpc_worker.go [hunk 2]
```diff
@@ -312,23 +312,36 @@ func (w *rpcWorker) getClusterConfigFromRemote(conn *bufio.ReadWriter, clusterCo
 	return rsp.GetGetClusterConfig(), nil
 }
 
-func (w *rpcWorker) callRPC(conn *bufio.ReadWriter, req *pdpb.Request) (*pdpb.Response, error) {
+func (w *rpcWorker) callRPC(conn *bufio.ReadWriter, req *pdpb.Request) (resp *pdpb.Response, err error) {
+	// Record some metrics.
+	start := time.Now()
+	label := convertName(req.GetCmdType().String())
+	defer func() {
+		if err == nil {
```
Comment (qiuyesuifeng) [inline pd-client/rpc_worker.go]: Is it OK to use err here? Seems err is a local variable.

File: pd-client/metrics.go (hunk 1 above)
Comment (huachaohuang) [inline pd-client/metrics.go]: But pd-client's tests depend on pd-server...
Maybe we can learn from other projects to figure out how we should do that later.

Comment (siddontang) [inline pd-client/metrics.go]: can we move to a common pkg?

File: pd-client/rpc_worker.go (hunk 2 above)
Comment (huachaohuang) [inline pd-client/rpc_worker.go]: It's OK, I learn it from [here](https://blog.golang.org/defer-panic-and-recover).

File: pd-client/metrics.go (hunk 1 above)
Comment (siddontang) [inline pd-client/metrics.go]: seem that we can create a util pkg and let client and server use it.

**Comment by siddontang** [comment]
LGTM

**Comment by huachaohuang** [comment]
@overvenus PTAL

File: pd-client/rpc_worker.go [hunk 3]
```diff
@@ -312,23 +313,36 @@ func (w *rpcWorker) getClusterConfigFromRemote(conn *bufio.ReadWriter, clusterCo
 	return rsp.GetGetClusterConfig(), nil
 }
 
-func (w *rpcWorker) callRPC(conn *bufio.ReadWriter, req *pdpb.Request) (*pdpb.Response, error) {
+func (w *rpcWorker) callRPC(conn *bufio.ReadWriter, req *pdpb.Request) (resp *pdpb.Response, err error) {
```
Comment (overvenus) [inline pd-client/rpc_worker.go]: Finally, found a way to deal with `err := foo()` :-)

**Comment by overvenus** [comment]
LGTM

**Pull request merged**